packages/backend/prisma/client.ts

```diff
@@ -1,6 +1,66 @@
 import { PrismaClient, Prisma } from '@prisma/client';
+import config from 'config';
+import gcm from 'helpers/gcmUtil';
 
 const prisma = new PrismaClient();
 
+// Prisma middleware to encrypt and decrypt data on fly
+prisma.$use(async (params, next) => {
+    if (params.model !== 'connections') {
+        return next(params);
+    }
+
+    const isReadOperation = ['findFirst', 'findMany'].includes(params.action);
+    const isWriteOperation = ['update', 'upsert'].includes(params.action);
+
+    if (isReadOperation) {
+        return handleReadOperation(params, next);
+    } else if (isWriteOperation) {
+        return handleWriteOperation(params, next);
+    }
+
+    return next(params);
+});
+
+async function handleReadOperation(
+    params: Prisma.MiddlewareParams,
+    next: (params: Prisma.MiddlewareParams) => Promise<any>
+): Promise<any> {
+    try {
+        const connections = await next(params);
+        if (Array.isArray(connections)) {
+            connections.forEach((connection) => {
+                connection.tp_customer_id = gcm.decrypt(connection.tp_customer_id, config.AES_SECRET);
```

Comment:
@ashutoshdhande I'm not too sure if we're handling other tables here to not have any encryption. Will this not break for other tables since they don't have the `tp_customer_id` field?

Similarly for the `handleWriteOperation` function. Can we add a filter to only do this for the `connection` table?

Reply:
Nvm I saw you're handing it above.